cli/interface.go

```diff
@@ -0,0 +1,215 @@
+		vm, releaseFunc, err := vmByContainerID(context, container)
+		if err != nil {
+			return err
+		}
+		defer releaseFunc()
+
+		ip := context.String("ip")
+		conf := &api.InterfaceDescription{
+			Name:    context.String("name"),
+			Ip:      ip,
+			Mac:     context.String("mac"),
+			TapName: context.String("tapname"),
+			Mtu:     context.Uint64("mtu"),
+		}
+
+		if err = vm.AddNic(conf); err != nil {
+			return err
+		}
+		fmt.Println("Add interface successfully.")
+		return nil
+	},
+}
+
+var infListCommand = cli.Command{
+	Name:      "ls",
+	Aliases:   []string{"list"},
+	Usage:     "list network interfaces in a container",
+	ArgsUsage: `ls <container-id>`,
+	Flags:     []cli.Flag{},
+	Action: func(context *cli.Context) error {
+		container := context.Args().First()
+		vm, releaseFunc, err := vmByContainerID(context, container)
+		if err != nil {
+			return err
+		}
+		defer releaseFunc()
+
+		tw := tabwriter.NewWriter(os.Stdout, 10, 1, 3, ' ', 0)
+		fmt.Fprintln(tw, "Name\tMac\tIP\tMtu")
+		nics := vm.AllNics()
+		for _, i := range nics {
+			fmt.Fprintf(tw, "%s\t%s\t%s\t%d\n", i.NewName, i.MacAddr, i.IpAddr, i.Mtu)
+		}
+		tw.Flush()
+		return nil
+	},
+}
+
+var infRmCommand = cli.Command{
+	Name:      "rm",
+	Aliases:   []string{"delete"},
+	Usage:     "remove an interface from container",
+	ArgsUsage: `rm <container-id> <interface-name>`,
+	Flags:     []cli.Flag{},
+	Action: func(context *cli.Context) error {
+		container := context.Args().First()
+		inf := context.Args().Get(1)
+		if inf == "" {
+			return cli.NewExitError("please specify an interface to delete", -1)
+		}
+		vm, releaseFunc, err := vmByContainerID(context, container)
+		if err != nil {
+			return err
+		}
+		defer releaseFunc()
+
+		nics := vm.AllNics()
+		for _, i := range nics {
+			if i.NewName == inf {
+				if err = vm.DeleteNic(i.Id); err != nil {
+					return cli.NewExitError(fmt.Sprintf("failed to delete interface %q: %v", inf, err), -1)
+				}
+				fmt.Printf("Interface %q is deleted\n", inf)
+				break
+			}
+		}
+		return nil
+	},
+}
+
+var infUpdateCommand = cli.Command{
+	Name:      "update",
+	Usage:     "update configuration of interface",
+	ArgsUsage: `update <container-id> <interface-name>`,
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "add-ip",
+			Usage: "add a new ip address with mask (format: 192.168.0.2/24)",
+		},
+		cli.StringFlag{
+			Name:  "delete-ip",
+			Usage: "add a new ip address with mask (format: 192.168.0.2/24)",
+		},
+		cli.IntFlag{
+			Name:  "mtu",
+			Usage: "update mtu",
+		},
+	},
+	Action: func(context *cli.Context) error {
+		container := context.Args().First()
+		vm, releaseFunc, err := vmByContainerID(context, container)
+		if err != nil {
+			return err
+		}
+		defer releaseFunc()
+
+		targetInf := context.Args().Get(1)
+		if targetInf == "" {
+			return cli.NewExitError("please specify an interface name to update", -1)
+		}
+
+		conf := &api.InterfaceDescription{
+			Id:   "-1",
+			Name: context.String("name"),
```

Comment:
What's the value of context.String("name")? is it existed?

Reply:
Oh, my fault, this should be `context.String("tapname"),`

I modify some logic in this upstream PR, and forgot to update tap setting logic.